### DISCUSSION TITLE
pluto tv 80s rewind channel freezing- xteve with streamlink through emby

### DISCUSSION THREAD

**Discussion by merrychrishna** [streamlink/streamlink#4697] [Q&A] [answered]
the script im using is plutoTV.py from here:
https://github.com/ReenigneArcher/web2tv

i use it with streamlink through xteve with this command in xteve:
--stdout --loglevel info [URL] --default-stream best

user agent is blank in xteve:
thats the only way streams will play

i must use streamlink with this script since streamlink stitches the commercials to the content perfectly, well, all channels except for 80s Rewind for some reason.

Describe the Bug
only the pluto tv 80s Rewind channel is freezing for me every time it goes to commercial. this is the reason i switched from the old pluto tv script to use your pluto tv script to be used with xteve and streamlink.
would you happen to know what i can try? the other channels make it through the commercials fine though.

Code Snippets
No response

Expected Behavior
the 80s channel should transition through the commercial and then continue playing the show

Additional Context
it seems to happen exactly before a commercial starts, is when it freezes forever. i never make it to the pluto tv screen saver commercial. but the channel plays fine up until the commercial.

also if i stop the channel and start it again, the channel plays fine up until the commercial again, then freezes again.

Operating System and Version
windows 10, regular android 9

Python Version
Python 3.7.9 and python 3.9

**Comment by bastimeyer** [comment, marked as answer]
> commercials

HLS stream discontinuities are unsupported.
#4535

You can try filtering out unwanted ad segments to perserve the output stream's presentation timestamps and avoid decoding/rendering freezes in your video player. This can be done via [`--hls-segment-ignore-names`](https://streamlink.github.io/cli.html#cmdoption-hls-segment-ignore-names) if you know which segment names you're looking for. Otherwise, you'll have to write a custom plugin with a custom HLS subclass that implements ad segment filtering for these streams.

Or just avoid using Streamlink and open the HLS URL that gets returned by the script you're using in your video player directly, if it supports reading HLS streams.

**Reply by merrychrishna** [reply]
where do i paste --hls-segment-ignore-names? do i put that in the streamlink config file?
 
also, i do not see the adds or commercials. they are already filtered out with teh plutotv.py script so all i see are like a screen saver that says PLUTO TV floating through space. so im not really sure what to type or where to type it. i also dont see stream discontinuity anywhere in the xteve console and i dont know where im supposed to find the streamlink logs location because streamlinkrc is not in the %appdata%streamlink folder

**Reply by merrychrishna** [reply]
where can i find the names? im sure they are somewhere but i dont really know what you mean by segment names

**Reply by merrychrishna** [reply]
there is also a plugin for pluto tv for streamlink here https://github.com/streamlink/streamlink/blob/master/src/streamlink/plugins/pluto.py but i dont know how to use it and is different from the script im using. the script im using creates an m3u and epg files.

**Reply by merrychrishna** [reply]
also i think you ignored that fact that i want to use streamlink with xteve through emby

**Reply by gravyboat** [reply]
> also i think you ignored that fact that i want to use streamlink with xteve through emby

This is outside the scope of any sort of support or suggestions we can provide. We're not going to assist people with combining random pieces of software. You'll need to figure out how to get them working on your own. We also can't and won't provide support for some random script from another repo which is your main use case. Bastimeyer has already provided all the help that we can in this scenario.

I'm going to lock this discussion now as there's nothing more we can do and it's way outside the scope of Streamlink.

**Reply by bastimeyer** [reply]
I didn't ignore anything here.

You are using a **custom script/application** which returns a single HLS URL that you are then plugging into Streamlink. [Streamlink does support standalone HLS URLs as input](https://streamlink.github.io/cli/protocols.html), but there is no custom logic that can be applied here for filtering out HLS segments other than the method I've posted earlier, which requires knowing the file names of unwanted segments, which means you'd have to take a look at the HLS playlist content first. And plugging an M3U8 URL into Streamlink resolved by a 3rd-party application doesn't mean that "ads are already filtered out". That's not how it works.

As you already pointed out, Streamlink does have a plugin for Pluto built-in, and it filters out ads, as you can see here:
https://github.com/streamlink/streamlink/blob/4.2.0/src/streamlink/plugins/pluto.py#L20-L24
Whether that's working reliably, I can't say, because I don't use that plugin. If it's not working correctly, then a plugin issue can be opened. However, you're not using this plugin and you're using the HLS-plugin instead, which handles regular M3U8 URLs.

If you don't know how to actually use Streamlink, then I suggest you take a look at the documentation.
https://streamlink.github.io/cli.html
https://streamlink.github.io/cli/tutorial.html

Either way, if you're using any custom applications, then you're on your own, and there won't be any support. This is not a general purpose tech support forum.
